cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -154,6 +156,7 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
           if(Python2_Development_FOUND)
             set(PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}")
             set(PYTHON_LIBRARIES "${Python2_LIBRARIES}")
+            set(PYTHON_LIBRARY_DIR "${Python2_LIBRARY_DIRS}")
```

Comment:
I think you are missing setting this variable further down, in https://github.com/root-project/root/pull/5771/commits/9511193de1526046eb18b3ae1a790cc65be0cee8#diff-5a4af40c5dab2eba7ad500714197987aR193

Reply:
Line 193